solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1324,16 +1348,28 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
     ) internal {
         ILiquidityPoolV1Converter converter = ILiquidityPoolV1Converter(payable(ownedBy(_poolToken)));
 
-        IERC20Token[] memory reserveTokens = new IERC20Token[](2);
-        uint256[] memory minReturns = new uint256[](2);
-        reserveTokens[0] = _reserveToken1;
-        reserveTokens[1] = _reserveToken2;
-        minReturns[0] = 1;
-        minReturns[1] = 1;
-
         // ensure that the contract can receive ETH
         updatingLiquidity = true;
-        converter.removeLiquidity(_poolAmount, reserveTokens, minReturns);
+
+        if (isStandardPool(converter)) {
+            IERC20Token[2] memory reserveTokens;
+            uint256[2] memory minReturns;
+            reserveTokens[0] = _reserveToken1;
+            reserveTokens[1] = _reserveToken2;
+            minReturns[0] = 1;
+            minReturns[1] = 1;
+            converter.removeLiquidity(_poolAmount, reserveTokens, minReturns);
+        } else {
+            IERC20Token[] memory reserveTokens = new IERC20Token[](2);
+            uint256[] memory minReturns = new uint256[](2);
+            reserveTokens[0] = _reserveToken1;
+            reserveTokens[1] = _reserveToken2;
+            minReturns[0] = 1;
+            minReturns[1] = 1;
+            converter.removeLiquidity(_poolAmount, reserveTokens, minReturns);
```

Comment:
Same here

Reply:
It is not, because `IERC20Token[2]` and `IERC20Token[]` are two different types with no common "ancestor".
Same goes for `uint256[2]` and `uint256[]`.
If the language supported templates, then we could unify these two blocks.
Otherwise, I don't see how this can be done.